File: testcases/x86_testcases/x86_output/test8/test8.c
#include <stdio.h>

// Define structures for Rectangle and Circle
typedef struct {
    int width;
    int height;
} Rectangle;

typedef struct {
    int radius;
} Circle;

// Function prototypes
int rectangle_area(Rectangle rect);
float circle_area(Circle circle);

int main() {
    // Creating instances of Rectangle and Circle
    Rectangle rect = {5, 4};
    Circle circle = {3};

    // Calculating areas and printing
    printf("Area of Rectangle: %d\n", rectangle_area(rect));
    printf("Area of Circle: %f\n", circle_area(circle));

    return 0;
}

// Function definitions
int rectangle_area(Rectangle rect) {
    return rect.width * rect.height;
}

float circle_area(Circle circle) {
    return 3.14 * circle.radius * circle.radius;
}

File: testcases/x86_testcases/test8.py
class Rectangle:
    def __init__(self, width: int, height: int):
        self.width : int = width
        self.height : int = height

    def area(self) -> int:
        return (self.width * self.height)

class Square:
    def __init__(self, side: int):
        self.side : int = side

    def area(self) -> int:
        return (self.side * self.side)

def main() -> None:
    # Creating object of class and calling there methods
    rect : Rectangle = Rectangle(5, 4)
    area_rect : int = rect.area()
    print("Area of Rectangle: ", area_rect)

    # Creating object of class and calling there methods
    square : Square = Square(3)
    area_square : int = square.area()
    print("Area of Square: ", area_square)

    return

if __name__ == "__main__":
    main()
